src/test/java/testsuite/integration/AwsIamAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package testsuite.integration;
+
+import com.mysql.cj.conf.PropertyKey;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.MethodOrderer.Alphanumeric;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import software.aws.rds.jdbc.mysql.Driver;
+
+@Disabled
+@TestMethodOrder(Alphanumeric.class)
+public class AwsIamAuthenticationIntegrationTest {
+
+    private static final String DB_CONN_STR_PREFIX = "jdbc:mysql:aws://";
+    private static final String DB_READONLY_CONN_STR_SUFFIX =
+        System.getenv("DB_READONLY_CONN_STR_SUFFIX");
+    private static final String TEST_DB_CLUSTER_IDENTIFIER =
+        System.getenv("TEST_DB_CLUSTER_IDENTIFIER");
+    private static final String TEST_PASSWORD = System.getenv("TEST_PASSWORD");
+    private static final String TEST_DB_USER = System.getenv("TEST_DB_USER");
+    private static final String DB_CONN_STR = DB_CONN_STR_PREFIX + TEST_DB_CLUSTER_IDENTIFIER + DB_READONLY_CONN_STR_SUFFIX;
+    private static Driver testDriver;
+
+
+    @BeforeEach
+    public void initTest() throws SQLException {
+        testDriver = new Driver();
+        DriverManager.registerDriver(testDriver);
+    }
+
+    @AfterEach
+    public void endTest() throws SQLException {
+        DriverManager.deregisterDriver(testDriver);
+    }
+
+    @Test
+    void invalidConnectionAwsIamAuthenticationWrongUser() {
+        Properties props = initProp("WRONG_" + TEST_DB_USER + "_USER", TEST_PASSWORD);
```

Comment:
Code -> Analyze code -> run inspection by name -> Local variable or parameter can be final -> set this file to scope -> fix all